File: android/designsystem/src/main/java/com/cyriltheandroid/travelcase/designsystem/badge/SmallBadge.kt
package com.cyriltheandroid.travelcase.designsystem.badge

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.cyriltheandroid.travelcase.designsystem.text.Text
import com.cyriltheandroid.travelcase.designsystem.text.TextIntent

@Composable
fun SmallBadge(
    text: String,
    modifier: Modifier = Modifier,
) {
    Box(
        modifier = modifier
            .size(16.dp)
            .background(Color.White, CircleShape),
        contentAlignment = Alignment.Center,
    ) {
        Text(
            text = text,
            intent = TextIntent.Small,
            textAlign = TextAlign.Center
        )
    }
}

File: iosApp/iosApp/designsystem/badge/SmallBadge.swift
import SwiftUI

struct SmallBadge: View {
    let text: String
    
    init(text: String) {
        self.text = text
    }
    
    var body: some View {
        Text(text: text, intent: .small)
            .frame(width: 16, height: 16, alignment: .center)
            .background(Color.white)
            .clipShape(Circle())
    }
}
